phac-nml/irida-next
# Group Attachments: Index and Upload

## What does this PR do and why?
This PR adds the attachments UI (for index listing and uploading) to groups

To follow:
Destroy will be added after [project attachments destroy](https://github.com/phac-nml/irida-next/pull/762) is merged.
Possible streamlining between group/project attachments (controller concerns etc.).

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/3106aab0-ad11-4781-b966-6ac2e956aa4b)

## How to set up and validate locally
1. Navigate to a group
3. Click "Files" on the sidebar
4. Verify `No Files` state
5. Verify the upload works by uploading a bunch of files, preferably with different formats and types
6. Verify per page selection, pagination, and sorting/filtering works as expected
7. Click on a filename and ensure it downloads

Other tests:
8. Verify project `guests` cannot see the `Files` sidebar link in groups
9. Verify project `analysts` cannot see the `Upload Files` button on the groups attachments index page
10. After uploading a file, wait ~1 minute to verify the table automatically updates the `Uploaded time_ago`
11. Verify sort, filter, and limit selection work as expected, and work together (ie: sort, then add a filter, then select a limit; all should still be enabled)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
